starters/remix/app/components/resolvers/types.tsx

```diff
@@ -0,0 +1,222 @@
+import {
+  AutoField,
+  ComponentConfig,
+  ComponentData,
+  Field as PuckField,
+} from '@measured/puck'
+
+export type Component = Omit<
+  ComponentConfig<any, any, Omit<ComponentData<any, string>, 'type'>>,
+  'type'
+>
+
+export type Field = PuckField & {
+  config?: {
+    fieldName?: string
+  }
+}
+
+export const fieldText: Field = {
+  type: 'text',
+}
+
+export const fieldTextArea: Field = {
+  type: 'textarea',
+}
+
+export const fieldLink: Field = {
+  type: 'object',
+  objectFields: {
+    title: { type: 'text' },
+    url: { type: 'text' },
+  },
+}
+
+export const fieldLinks: Field = {
+  type: 'array',
+  arrayFields: fieldLink.objectFields,
+}
+
+export const fieldMediaExternal: Field = {
```

Comment:
This should not be some dynamic code; instead, the values?
The files are volatile data; don't know if this is correct

Reply:
Agree on this and nice catch, since this is an Editor feature is WIP and for testing purposes